text/0000-convert-id.md

```diff
@@ -0,0 +1,203 @@
+identity conversion, it is less clear that this was intentional.
+With `identity`, this intent becomes clearer.
+
+## The `drop` function as a precedent
+
+The `drop` function in `core::mem` is defined as `pub fn drop<T>(_x: T) { }`.
+The same effect can be achieved by writing `{ _x; }`. This presents us
+with a precendent that such trivial functions are considered useful and
+includable inside the standard library even though they can be written easily
+inside a user's crate.
+
+## Avoiding repetition in user crates
+
+Here are a few examples of the identity function being defined and used:
+
++ https://docs.rs/functils/0.0.2/functils/fn.identity.html
++ https://docs.rs/tool/0.2.0/tool/fn.id.html
++ https://github.com/hephex/api/blob/ef67b209cd88d0af40af10b4a9f3e0e61a5924da/src/lib.rs
+
+There's a smattering of more examples. To reduce duplication, it
+should be provided in the standard library as a common place it is defined.
+
+## Precedent from other languages
+
+There are other languages that include an identity function in
+their standard libraries, among these are:
+
++ [Haskell](http://hackage.haskell.org/package/base-4.10.1.0/docs/Prelude.html#v:id), which also exports this to the prelude.
++ [Scala](https://www.scala-lang.org/api/current/scala/Predef$.html#identity[A](x:A):A), which also exports this to the prelude.
++ [Java](https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html#identity--), which is a widely used language.
++ [Idris](https://www.idris-lang.org/docs/1.0/prelude_doc/docs/Prelude.Basics.html), which also exports this to the prelude.
++ [Ruby](http://ruby-doc.org/core-2.5.0/Object.html#method-i-itself), which exports it to what amounts to the top type.
++ [Racket](http://docs.racket-lang.org/reference/values.html)
++ [Julia](https://docs.julialang.org/en/release-0.4/stdlib/base/#Base.identity)
++ [R](https://stat.ethz.ch/R-manual/R-devel/library/base/html/identity.html)
++ [F#](https://msdn.microsoft.com/en-us/visualfsharpdocs/conceptual/operators.id%5B%27t%5D-function-%5Bfsharp%5D)
++ [Clojure](https://clojuredocs.org/clojure.core/identity)
++ [Agda](http://www.cse.chalmers.se/~nad/repos/lib/src/Function.agda)
++ [Elm](http://package.elm-lang.org/packages/elm-lang/core/latest/Basics#identity)
+
+## The case for inclusion in the prelude
+
+Let's compare the effort required, assuming that each letter
+typed has a uniform cost wrt. effort.
+
+```rust
+use std::convert::identity; iter.filter_map(identity)
+
+fn identity<T>(x: T) -> T { x } iter.filter_map(identity)
+
+iter.filter_map(::std::convert::identity)
+
+iter.filter_map(identity)
+```
+
+Comparing the length of these lines, we see that there's not much difference in
+length when defining the function yourself or when importing or using an absolute
+path. But the prelude-using variant is considerably shorter. To encourage the
+use of the function, exporting to the prelude is therefore a good idea.
```

Comment:
> But the prelude-using variant is considerably shorter.

This is true for literally every possible library addition, so I don't think its a good argument.